transport.go

```diff
@@ -3,45 +3,140 @@ package libp2praft
 import (
 	"context"
 	"errors"
+	"fmt"
+	"io"
 	"log"
 	"net"
-	"strings"
 	"time"
 
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/raft"
+	logging "github.com/ipfs/go-log/v2"
+	gostream "github.com/libp2p/go-libp2p-gostream"
 	"github.com/libp2p/go-libp2p/core/host"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/core/protocol"
-
-	"github.com/hashicorp/raft"
-	gostream "github.com/libp2p/go-libp2p-gostream"
 )
 
 const RaftProtocol protocol.ID = "/raft/1.0.0/rpc"
 
-// This provides a custom logger for the network transport
-// which intercepts messages and rewrites them to our own logger
-type logForwarder struct{}
-
-var logLogger = log.New(&logForwarder{}, "", 0)
-
-// Write forwards to our go-log logger.
-// According to https://golang.org/pkg/log/#Logger.Output
-// it is called per line.
-func (fw *logForwarder) Write(p []byte) (n int, err error) {
-	t := strings.TrimSuffix(string(p), "\n")
-	switch {
-	case strings.Contains(t, "[DEBUG]"):
-		logger.Debug(strings.TrimPrefix(t, "[DEBUG] raft-net: "))
-	case strings.Contains(t, "[WARN]"):
-		logger.Warn(strings.TrimPrefix(t, "[WARN]  raft-net: "))
-	case strings.Contains(t, "[ERR]"):
-		logger.Error(strings.TrimPrefix(t, "[ERR] raft-net: "))
-	case strings.Contains(t, "[INFO]"):
-		logger.Info(strings.TrimPrefix(t, "[INFO] raft-net: "))
+var raftLogger = logging.Logger("raftlib")
+
+// this implements github.com/hashicorp/go-hclog
+type hcLogToLogger struct {
```

Comment:
Can you please export this type? (It will require to add comments to the exported methods for checks to pass).

Reply:
updated, have a try plz